frigate/events/cleanup.py

```diff
@@ -197,9 +199,17 @@ def run(self) -> None:
             self.expire(EventCleanupType.snapshots)
 
             # drop events from db where has_clip and has_snapshot are false
-            delete_query = Event.delete().where(
-                Event.has_clip == False, Event.has_snapshot == False
+            events = (
+                Event.select()
+                .where(Event.has_clip == False, Event.has_snapshot == False)
+                .iterator()
             )
-            delete_query.execute()
+            events_to_delete = [e.id for e in events]
+            if len(events_to_delete) > 0:
+                Event.delete().where(Event.id << events_to_delete).execute()
+
+                if self.config.semantic_search.enabled:
```

Comment:
This is added to ensure the embeddings database is kept in sync with the events table and we remove things based on the same retention policies.